lib/common/xml.c

```diff
@@ -2258,23 +2258,47 @@ find_xml_node(xmlNode * root, const char *search_path, gboolean must_find)
     return NULL;
 }
 
-xmlNode *
-find_entity(xmlNode * parent, const char *node_name, const char *id)
+/* As the name suggests, the perfect match is required for both node
+   name and fully specified attribute, otherwise, when attribute not
+   specified, the outcome is the first node matching on the name. */
+static xmlNode *
+find_entity_by_attr_or_just_name(xmlNode *parent, const char *node_name,
+                                 const char *attr_n, const char *attr_v)
 {
-    xmlNode *a_child = NULL;
+    xmlNode *child, *first_child = NULL;
 
-    for (a_child = __xml_first_child(parent); a_child != NULL; a_child = __xml_next(a_child)) {
-        /* Uncertain if node_name == NULL check is strictly necessary here */
-        if (node_name == NULL || strcmp((const char *)a_child->name, node_name) == 0) {
-            const char *cid = ID(a_child);
-            if (id == NULL || (cid != NULL && strcmp(id, cid) == 0)) {
-                return a_child;
+    /* ensure attr_v specified when attr_n is */
+    CRM_CHECK(attr_n == NULL || attr_v != NULL, return NULL);
+
+    for (child = __xml_first_child(parent); child != NULL; child = __xml_next(child)) {
+        /* XXX uncertain if the first check is strictly necessary here */
+        if (node_name == NULL || !strcmp((const char *) child->name, node_name)) {
+            if (attr_n == NULL
+                    || crm_str_eq(crm_element_value(child, attr_n), attr_v, TRUE)) {
+                return child;
             }
+            first_child = child;
```

Comment:
Did you mean to wrap this with if (first_child == NULL)?

Reply:
Heck, indeed, good catch.
The subfix (amendment of the forgotten piece exposed directly in the tests)
that itself went too carelessly wrt. intended semantics.

Reply:
In fact, it wasn't meant to be present at all (Friday + mess arising from several branches
worked on in parallel?)

Reply:
Of course, now it might make sense to put it all into one big `if`.